compose/config/config.py

```diff
@@ -363,7 +363,8 @@ def process_config_file(config_file, service_name=None):
     config_file = config_file._replace(config=processed_config)
     validate_against_fields_schema(config_file)
 
-    if service_name and service_name not in processed_config:
+    if service_name and (config_file.version == 1 and service_name not in processed_config or
+                         config_file.version == 2 and service_name not in processed_config['services']):
```

Comment:
Instead of branching again on the version, could we set a `services` variable to the correct value in the  branch above this?

```python
processed_config['services'] = services = interpolated_config

...

processed_config = services = interpolated_config

```

Then this can be just 

```python

if service_name and service_name in services:
    ...

```